src/lib/cli.rs

```diff
@@ -135,6 +140,8 @@ fn run(cli_args: CliArgs, global_config: &GlobalConfig) {
     } else {
         runner::run(config, &task, env_info, &cli_args);
     }
+
+    recursion_level::decrease_level();
```

Comment:
How would you take care of panics here? `panic::catch_unwind` doesn't seem enough (at least in the test suite).

Reply:
it doesn't matter if it panics and there is no need to decrease level.
child processes can't change parent process env.
so you just increment. when it ends, parent is unaffected.

Reply:
Oof, you're absolutely right, of course! Still had the mindset of the previous solution where a decrease was necessary, sorry!

Will fix this tomorrow accordingly :)